test/unit/vagrant/machine_test.rb

```diff
@@ -793,6 +793,13 @@ def detect?(machine)
         it "paranoid should be default" do
           expect(instance.ssh_info[:paranoid]).to be(false)
         end
+        it "extra_args should be nil" do
+          expect(instance.ssh_info[:extra_args]).to be(nil)
+        end
+        it "extra_args should be set" do
+          instance.config.ssh.extra_args = ["-L" "127.1.2.7:8008:127.1.2.7:8008"]
+          expect(instance.ssh_info[:extra_args]).to eq(["-L" "127.1.2.7:8008:127.1.2.7:8008"])
+        end
```

Comment:
Include test with string value being set into config that validates proper conversion to array